recipe_cuda.Dockerfile

```diff
@@ -0,0 +1,103 @@
+        mkdir -p /usr/local/share/just/user_run_patch/; \
+        echo "echo 'Cuda recipe is out of date. Consdier PRing to update CUDA_REPO_REF' >&2" > /usr/local/share/just/user_run_patch/00_cuda_outdated_warning; \
+        echo "echo 'Consider updating CUDA_REPO_REF(${CUDA_REPO_REF}) to ${new_master_ref} in a PR' >&2" >> /usr/local/share/just/user_run_patch/00_cuda_outdated_warning; \
+        echo 'echo "Or delete $0"' >> /usr/local/share/just/user_run_patch/00_cuda_outdated_warning; \
+        chmod 755 /usr/local/share/just/user_run_patch/00_cuda_outdated_warning; \
+      fi; \
+    fi; \
+    apk add --no-cache --virtual .deps curl ca-certificates; \
+    curl -fsSLO https://gitlab.com/nvidia/container-images/cuda/-/archive/${CUDA_REPO_REF}/cuda.tar.gz; \
+    tar xf /cuda.tar.gz; \
+    mv /cuda-* /cuda; \
+    rm /cuda.tar.gz; \
+    apk del .deps
+
+ADD 00_cuda_sanity_check /usr/local/share/just/root_run_patch/
+ADD 10_load_cuda_env /usr/local/share/just/user_run_patch/
+ADD 30_ldconfig 30_install_cuda /usr/local/share/just/container_build_patch/
+RUN chmod 755 /usr/local/share/just/root_run_patch/00_cuda_sanity_check \
+              /usr/local/share/just/container_build_patch/30_ldconfig \
+              /usr/local/share/just/container_build_patch/30_install_cuda; \
+    chmod 644 /usr/local/share/just/user_run_patch/10_load_cuda_env
+
+ONBUILD ARG CUDA_VERSION=11.7.0
+ONBUILD ARG CUDNN_VERSION=
+ONBUILD ARG CUDA_RECIPE_TARGET=runtime
+
+ONBUILD RUN set -o pipefail; \
+            apk add --no-cache --virtual .deps curl ca-certificates; \
+            mkdir -p /usr/local/share/just/info/cuda/keys; \
+            function parse_envvar(){ \
+              # Uses an awk trick to only find the unique names, based on the
+              # second column, which in this case the environment variable name.
+              # This works how we want because the x86_64 variables always come
+              # first in the cuda Dockerfiles.
+              awk '!x[$2]++' "${1}" | sed -nE '/^ENV/s/ENV ([^ ]*) /\1=/p'; \
+            }; \
+            function parse_os(){ \
+              parse_envvar "${1}/base/Dockerfile" > "/usr/local/share/just/info/cuda/10_cudaenv_${2}"; \
+              parse_envvar "${1}/runtime/Dockerfile" > "/usr/local/share/just/info/cuda/12_cudaenv_${2}"; \
+              parse_envvar "${1}/devel/Dockerfile" > "/usr/local/share/just/info/cuda/14_cudaenv_${2}"; \
+              if [ -n "${CUDNN_VERSION:+set}" ]; then \
+                # If this errors during build, you probably set CUDNN_VERSION wrong
+                parse_envvar "${1}/runtime/cudnn${CUDNN_VERSION}/Dockerfile" > "/usr/local/share/just/info/cuda/13_cudaenv_${2}"; \
+                parse_envvar "${1}/devel/cudnn${CUDNN_VERSION}/Dockerfile" > "/usr/local/share/just/info/cuda/13_cudaenv_${2}"; \
+              fi; \
+            }; \
+            # Env Vars
+            # Currently all the fedoras use the same vars, so the first one is good enough
```

Comment:
what does that mean, "the first one"

Reply:
you mention that Fedora is not supported in the REAME. should this say all redhats